lib/src/emitter.dart

```diff
@@ -535,6 +536,38 @@ class DartEmitter extends Object
     return out;
   }
 
+  @override
+  StringSink visitRecordType(RecordType spec, [StringSink? output]) {
+    final out = (output ??= StringBuffer())..write('(');
+    visitAll<Reference>(spec.positionalFieldTypes, out, (spec) {
+      spec.accept(this, out);
+    });
+    if (spec.namedFieldTypes.isNotEmpty) {
+      if (spec.positionalFieldTypes.isNotEmpty) {
+        out.write(', ');
+      }
+      out.write('{');
+      visitAll<MapEntry<String, Reference>>(spec.namedFieldTypes.entries, out,
+          (entry) {
+        out.write('${entry.key} ');
```

Comment:
These are backwards. See the illegal syntax generated in `record_type_test.dart`: `({named int, other List<int>})`

Reply:
Good catch! That's how you write bad code _and_ bad tests ;) Fixed.